source/src/protocols/recon_design/recon_util.cc

```diff
@@ -45,31 +45,40 @@ get_designable_residues( core::pose::Pose & pose, std::string resfile ) {
 
 /// Based on a pose and the indices of all designable residues, get a
 /// string of all designable AAs concatenated
-std::string
+utility::vector1< std::string >
 get_designable_sequence ( core::pose::Pose & pose, utility::vector1< core::Size > designable_residues ) {
-	std::string sequence = "";
+	//std::string sequence = "";
+    utility::vector1< std::string > sequence;
 	for ( core::Size seqpos: designable_residues ) {
-		sequence += pose.residue( seqpos ).name1();
+		//sequence += pose.residue( seqpos ).name1();
+        sequence.push_back(pose.residue_type( seqpos ).base_name());
```

Comment:
One of the issues you'll run into here is that it looks like ResidueTypeConstraint takes a string which it compares with the `name3()` of the residue types. In general, the `base_name()` is not going to be equal to the `name3()`. (It is equal for standard residues, but isn't necessarily for noncanonicals and other residues which don't have a simple one letter code -> base_name() correspondence.)

(My recommendation is to either use `name3()` here, or add a mode to ResidueTypeConstraint which works on `base_name()`.)